hera_mc/cm_hookup.py

```diff
@@ -287,14 +287,20 @@ def __hookup_add_correlator_levels(self, hookup_dict):
         pf_input = []
         for k in sorted(hookup_dict['hookup'].keys()):
             for p in hookup_dict['hookup'][k].keys():
-                f_engine = hookup_dict['hookup'][k][p][-1].downstream_part
+                try:
+                    f_engine = hookup_dict['hookup'][k][p][-1].downstream_part
+                except IndexError:
+                    continue
```

Comment:
Some people like it, but I personally am not a big fan of this approach to bounds-checking. In this case in particular, it is not obvious which indexing operation is the one that can fail. Can you rewrite using an `if` statement with a `break` if a usable record is found?

Reply:
OK - so you are going to make me understand what's going on again eh? ;-)  I've fixed this and below (and found a potential silent error).